### PR TITLE
Ingest sheet simplified with preview and facetlinks

### PR DESCRIPTION
**Pull request by katdivyareddy10**
Screen Shot 2020-10-06 at 12 11 53 PM

### CODE REVIEW

**Review by adamjarling — CHANGES_REQUESTED** [review]
A few design requests...

image

File: assets/js/components/BatchEdit/PreviewItems.js [hunk 1]
```diff
@@ -32,41 +32,34 @@ const previewItem = css`
...
+                      : representativeFileSet.url
+                      ? `${representativeFileSet.url}/square/128,128/0/default.jpg`
+                      : "https://bulma.io/images/placeholders/128x128.png"
+                  }
+                />
+              </figure>
+            </Link>
+          </li>
+        ))}
+      </ul>
```
Comment (adamjarling) [inline assets/js/components/BatchEdit/PreviewItems.js:57]: How about we extract this component to the `/UI` directory now that we're using it in 2 different places.

**Comment by divyareddynu** [comment]
@adamjarling I pushed changes.
Screen Shot 2020-10-07 at 9 21 18 AM

File: assets/js/components/UI/Image.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
```
Comment (adamjarling) [inline assets/js/components/UI/Image.jsx:1]: Can we change the title of this file to `assets/js/components/UI/WorkImage.jsx`?

File: assets/js/components/UI/Image.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
```
Comment (adamjarling) [inline assets/js/components/UI/Image.test.js:1]: Same here, file title to `assets/js/components/UI/WorkImage.test.js`

File: assets/js/components/UI/PreviewItems.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+const previewItem = css`
+  opacity: 1;
+  transition: 0.3s;
+
+  &:hover {
+    opacity: 0.75;
+  }
+`;
+
+export default function PreviewItems({ items = [] }) {
```
Comment (adamjarling) [inline assets/js/components/UI/PreviewItems.js:37]: Following our naming convention pattern, this should be `export default function UIPreviewItems({ items = [] }) {`

File: assets/js/components/UI/PreviewItems.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+              </figure>
+            </Link>
+          </li>
+        ))}
+      </ul>
+    </div>
+  );
+}
+
+PreviewItems.propTypes = {
```
Comment (adamjarling) [inline assets/js/components/UI/PreviewItems.js:58]: Same here, `UIPreviewitems.propTypes = ...`

File: assets/js/components/UI/PreviewItems.test.js [hunk 6]
```diff
@@ -1,13 +1,11 @@
 import React from "react";
-import BatchEditPreviewItems from "./PreviewItems";
+import PreviewItems from "./PreviewItems";
```
Comment (adamjarling) [inline assets/js/components/UI/PreviewItems.test.js:2]: And then when we import it, `import UIPreviewItems from "./UIPreviewItems";`  ...and we'll continue the new variable title throughout the code.

File: assets/js/components/Work/CardItem.jsx [hunk 7]
```diff
@@ -1,7 +1,12 @@
 import React, { useState } from "react";
 import PropTypes from "prop-types";
 import { Link } from "react-router-dom";
-import { setVisibilityClass, formatDate } from "../../services/helpers";
+import {
+  setVisibilityClass,
+  formatDate,
+  getImageUrl,
+} from "../../services/helpers";
```
Comment (adamjarling) [inline assets/js/components/Work/CardItem.jsx:8]: Again, not a huge deal, but we can start using this pattern for aliasing relative paths when importing nested components:

```
import {
  setVisibilityClass,
  formatDate,
  getImageUrl,
} from "@js/services/helpers";
```

File: assets/js/services/helpers.js [hunk 8]
```diff
@@ -42,6 +42,13 @@ export function getClassFromIngestSheetStatus(status) {
   return "";
 }
 
+export function getImageUrl(representativeImage) {
+  if (typeof representativeImage === "object") {
+    return representativeImage.url || "";
+  }
+  return representativeImage;
+}
+
```
Comment (adamjarling) [inline assets/js/services/helpers.js:51]: Can we add a test for this function?  What happens if `representativeImage` is undefined, or an empty string?   It'd be nice if this returned an empty string as a default.

**Review by adamjarling — CHANGES_REQUESTED** [review]

**Comment by divyareddynu** [comment]
@adamjarling it looks like you may have reviewed an older version or my changes did not pull through. I will add using `@js` notation and push a PR in a bit. Can you delete your branch and re-import it?

**Comment by adamjarling** [comment]
Ooops, sorry about that.
